test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/PropertiesTest.scala

```diff
@@ -173,4 +175,285 @@ class PropertiesTest {
+
+    val prop2 = loadByteArrayOutputStream(out1)
+    assertAll(expected = prop1, actual = prop2)
+    // Avoid variable Date output which is last line in comment
+    // Matches JVM output
+    val commentsWithoutDate =
+      """|#A Header
+         |#Line2
+         |#Line3
+         |#Line4
+         |!AfterExclaim
+         |#AfterPound
+         |#Wow!""".stripMargin
+    assertTrue(out1.toString().startsWith(commentsWithoutDate))
+  }
+
+  @Test def checkPropertiesFormattedCorrectly(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property C", "see")
+    val out1 = storeStream(prop1)
+    assertTrue(out1.toString().trim().endsWith("""Property\ C=see"""))
+
+    val prop2 = new Properties()
+    prop2.put("Property B", "b ee#!=:")
+    val out2 = storeStream(prop2)
+    assertTrue(out2.toString().trim().endsWith("""Property\ B=b ee\#\!\=\:"""))
+
+    val prop3 = new Properties()
+    prop3.put("Property A", " aye\\\f\t\n\r\b")
+    val out3 = storeStream(prop3)
+    // JVM outputs \b as \u0008 and you can't just add \u0008 to the end
+    val result = new StringBuilder("""Property\ A=\ aye\\\f\t\n\r""")
+      .appendAll(Array('\\', 'u', '0', '0', '0', '8'))
+    assertTrue(out3.toString().trim().endsWith(result.toString))
+  }
+
+  @Test def storeWriterComments(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property A", " aye\\\f\t\n\r\b")
+    prop1.put("Property B", "b ee#!=:")
+    prop1.put("Property C", "see")
+    val out = storeWriter(prop1, "A Header")
+    val prop2 = loadByteArrayOutputStream(out)
+    assertAll(prop1, prop2)
+  }
+
+  // helper functions
+
+  def storeStream(props: Properties,
+      header: String = ""): ByteArrayOutputStream = {
+    val out = new ByteArrayOutputStream()
+    props.store(out, header)
+    out
+  }
+
+  def storeWriter(props: Properties,
+      header: String = ""): ByteArrayOutputStream = {
+    val out = new ByteArrayOutputStream()
+    props.store(new OutputStreamWriter(out), header)
+    out.close()
+    out
+  }
+
+  def loadByteArrayOutputStream(out: ByteArrayOutputStream): Properties = {
+    val props = new Properties()
+    props.load(new ByteArrayInputStream(out.toByteArray))
+    props
+  }
+
+  def assertAll(expected: Properties, actual: Properties): Unit = {
+    val e = expected.propertyNames()
+    while (e.hasMoreElements) {
+      val nextKey = e.nextElement().asInstanceOf[String]
+      assertEquals(actual.getProperty(nextKey), expected.getProperty(nextKey))
+    }
+  }
+
+  def loadStream(in: String): Properties = {
+    val prop = new java.util.Properties()
+    prop.load(new ByteArrayInputStream(
+        in.getBytes(StandardCharsets.ISO_8859_1)))
+    prop
+  }
+
+  def loadReader(in: String): Properties = {
+    val prop = new java.util.Properties()
+    prop.load(new InputStreamReader(new ByteArrayInputStream(
+        in.getBytes(StandardCharsets.UTF_8))))
+    prop
+  }
+
+  // scalastyle:off
+  lazy val filestr =
```

Comment:
```suggestion
  lazy val filestr = {
```
(and the closing `}` on the other side)

Reply:
This? 
>Every expression that spans more than one line should be enclosed in braces

Reply:
Yes, exactly.